### DISCUSSION TITLE
Mode automatique et contenu dynamique

### DISCUSSION THREAD

**Discussion by florianbouvot** [AmauriC/tarteaucitron.js#630] [Q&A]
Bonjour,

Je souhaiterais en savoir plus sur le Mode automatique (experimental) disponible dans la partie PRO.
Malgré mes recherches (site officiel, github), je n'ai pas trouvé d'information sur le fonctionnement.

Je test actuellement le service Tarte au Citron et je m'interroge concernant la mise en place de l'aspect dynamique.
Dans nos CMS le client peut ajouter des modules permettant d'intégrer des vidéos YouTube (ou autres contenus embarqués), est-ce que ce mode permet de détecter automatiquement les ID ?

<img width="959" alt="Capture d’écran 2021-03-24 à 10 26 47" src="https://user-images.githubusercontent.com/2061540/112287411-a4047e00-8c8c-11eb-8dc3-678f2befe128.png">  

Merci d'avance pour vos infos

**Comment by valentin-anamorphik** [comment]
Bonjour,

Vous souhaitez ajouter de manière global le service Youtube pour votre site et non gérer des "services Tarte au Citron" vidéo par vidéo (ID) ?

**Reply by florianbouvot** [reply]
Oui, exactement.

J'ai l'impression quand validant le service sans aucune info, ça fonctionne côté site.

**Reply by valentin-anamorphik** [reply]
Sans mentionner un ID depuis l'interface Tarte Au Citron ?

**Reply by florianbouvot** [reply]
Oui, j'active le service YouTube en ne remplissant aucun paramètres dans l'interface PRO Tarte au Citron puis je mets en place une div en respectant la documentation et ça fonctionne. 

Mais je voudrais pas mettre en place une logique bancale pour éviter les soucis de maintenance dans le temps.

**Reply by florianbouvot** [reply]
Y a t-il une meilleur approche ? ping @AmauriC

**Reply by AmauriC** [reply]
Déclarer le service youtube et ajouter la div du mode 'gratuit' fonctionne bien.

Le mode auto cherche et bloque avant consentement les balises js et iframes correpondant aux services supportés par tarteaucitron ; )

**Reply by florianbouvot** [reply]
@AmauriC Merci pour ton retour, je vais tester l'automatique pour voir ce que ça donne.

**Reply by valentin-anamorphik** [reply]
@AmauriC @florianbouvot 

J'ai activé le mode manuel et ajouté une div "vide" en déclarant le service youtube sur l'admin PRO TAC.

`<div id="tac-youtube" style="display:none;" aria-hidden="true"></div>`

J'ai bien une demande de consentement via la fenêtre modale TAC mais l'iframe YouTube est toujours visible peu import le choix de l'utilisateur pour les cookies Youtube.

**Comment by florianbouvot** [comment]
@valentin-anamorphik moi j'ai procéder ainsi :

1 - ajout du service "vide" dans l'espace PRO TaC

2 - ajout du code pour mon module qui permet l'ajout d'iframe YouTube : 

```
<div class="youtube_player" videoID="video_id" theme="dark" rel="1" controls="1" showinfo="1" autoplay="0" mute="0"></div>
```

La partie `videoID` est a personnalisée (dans mon cas la donnée vient du CMS). 

3 - gestion du ratio via CSS

La on sort du sujet mais j'utilise un composant CSS qui me permet de créer une zone au ratio 16:9

```
<div class="youtube_player c-ratio" videoID="video_id" theme="dark" rel="1" controls="1" showinfo="1" autoplay="0" mute="0"></div>
```

Je pourrais entrée plus dans le détail si cela t'intéresse ou partager un exemple réel (manque de temps ce matin).

**Reply by valentin-anamorphik** [reply]
Merci @florianbouvot 

1 - ajout du service "vide" dans l'espace PRO TaC
> Tu laisses vide le champ "Entrez la valeur de id" également sur l'espace PRO TaC ?

**Reply by florianbouvot** [reply]
Si j'avais une seule vidéo statique, j'ajouterai l'ID (cas d'un mini-site).

Dans un cas "réel" ou l'utilisateur ajoute librement des vidéos je laisse vide et tout est croisée avec mon CMS. 
Mais attention tu n'as ptet pas le même fonctionnement que moi.

**Reply by valentin-anamorphik** [reply]
Oui oui bien sûr ! Le fonctionnement me semble identique.

Ca semble fonctionner en mode manuel.
Dommage que l'iframe youtube ne soit pas remplacé par un message lorsque l'utilisateur refuse les cookies. Pour ma part, j'ai une div avec aucun contenu. L'utilisateur n'a donc pas connaissance de la présence d'une vidéo à cet endroit.

**Reply by florianbouvot** [reply]
J'ai commencé a utiliser TaC cette semaine donc je n'ai pas encore tout testé / finalisé.

Mais je pense qu'il est facile d'ajuster cela, je te ferais un retour semaine prochaine quand j'aurais repris.
